contracts/data/BinaryHeap.sol

```diff
@@ -0,0 +1,339 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.8;
+
+/**
+ * @title Binary Heap implementation
+ * @dev The data structure is configured as a max-heap
+ */
+
+library BinaryHeap {
+    error BinaryHeap__IndexOutOfBounds();
```

Comment:
I think we can remove this error and fall back to the default panic opcode.

Reply:
https://github.com/solidstate-network/solidstate-solidity/pull/143/commits/c692368e9a52ba3a541db97b4d8fe1966192d2a6